src/components/calculator.js

```diff
@@ -1,42 +1,60 @@
+import { useState } from 'react';
 import CalcBtn from './calcBtn';
 import CalcDisplay from './calcDisplay';
 import '../css/calculator.css';
+import calculate from './logic.js/calculate';
 
-const Calculator = () => (
-  <section id="calculator">
-    <div className="calc-row">
-      <CalcDisplay />
-    </div>
-    <div className="calc-row">
-      <CalcBtn text="AC" />
-      <CalcBtn text="+/-" />
-      <CalcBtn text="%" />
-      <CalcBtn text="÷" />
-    </div>
-    <div className="calc-row">
-      <CalcBtn text="7" />
-      <CalcBtn text="8" />
-      <CalcBtn text="9" />
-      <CalcBtn text="x" />
-    </div>
-    <div className="calc-row">
-      <CalcBtn text="4" />
-      <CalcBtn text="5" />
-      <CalcBtn text="6" />
-      <CalcBtn text="-" />
-    </div>
-    <div className="calc-row">
-      <CalcBtn text="1" />
-      <CalcBtn text="2" />
-      <CalcBtn text="3" />
-      <CalcBtn text="+" />
-    </div>
-    <div className="calc-row">
-      <CalcBtn extraClass="col-2" text="0" />
-      <CalcBtn text="." />
-      <CalcBtn text="=" />
-    </div>
-  </section>
-);
+const Calculator = () => {
+  const [data, setData] = useState({});
+
+  const handleClick = (text) => {
+    setData(calculate(data, text));
+  };
+
+  const getDisplayText = () => {
+    if (data) {
+      if (!data.next && !data.operation) return data.total ?? '0';
+      return data.next ?? '0';
+    }
+    return '0';
+  };
```

Comment:
- [ Optional ] Good job for implementing the logic, but for user-interaction I recommend you to preserve data(number) on screen because when i clicked on one of the operation(/,*,+,-) it change number to zero until I clicked another number and equal sign to see the result, it's little bit confuse for user-interaction . Kindly Fix it 
- For example as shown this [video](https://drive.google.com/file/d/13B_qgHPRXA8KaXbiafdH0TB3w7is533E/view?usp=share_link) i want to perform division of 6/2=3 , the flow of your implementation on screen display is 6->/ ->0->2->=->3 every time i clicked on the one of the operation it return to 0, So kindly fix it.

Reply:
Thank you for bringing that up, I will correct it